gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/api/use_case/cockpit/DeployModelToApiUpdateUseCase.java

```diff
@@ -0,0 +1,134 @@
+        OAIDomainService oaiDomainService,
+        UpdateApiDomainService updateApiDomainService,
+        DeployModelToApiDomainService deployModelToApiDomainService,
+        PlanCrudService planCrudService,
+        PageCrudService pageCrudService
+    ) {
+        this.oaiDomainService = oaiDomainService;
+        this.updateApiDomainService = updateApiDomainService;
+        this.deployModelToApiDomainService = deployModelToApiDomainService;
+        this.planCrudService = planCrudService;
+        this.pageCrudService = pageCrudService;
+    }
+
+    public DeployModelToApiUpdateUseCase.Output execute(DeployModelToApiUpdateUseCase.Input input) {
+        var organizationId = input.auditInfo().organizationId();
+        var environmentId = input.auditInfo().environmentId();
+        var importSwaggerDescriptorEntity = deployModelToApiDomainService.configure(input.mode(), input.swaggerDefinition());
+        var importDefinition = oaiDomainService.convert(organizationId, environmentId, importSwaggerDescriptorEntity, true, false);
+
+        managePlan(input);
+        manageDocumentationPage(input, importDefinition.getPages().getFirst());
+
+        importDefinition.getApiExport().setId(input.apiId());
+        importDefinition.getApiExport().setCrossId(input.apiCrossId());
+        importDefinition.getApiExport().setLabels(input.labels());
+
+        var apiUpdated = updateApiDomainService.updateV4(
+            ApiModelFactory.fromApiExport(importDefinition.getApiExport(), environmentId),
+            input.auditInfo()
+        );
+        var api = deployModelToApiDomainService.manageApiState(apiUpdated, input.auditInfo(), input.mode());
+
+        return new DeployModelToApiUpdateUseCase.Output(api);
+    }
+
+    private void managePlan(final Input input) {
+        final var plans = planCrudService.findByApiId(input.apiId());
+
+        if (plans.stream().noneMatch(plan -> plan.getPlanStatus() == PlanStatus.PUBLISHED)) {
+            planCrudService.create(
+                new Plan(input.apiId(), deployModelToApiDomainService.createDefaultPlan())
+                    .toBuilder()
+                    .needRedeployAt(Date.from(TimeProvider.now().toInstant()))
+                    .validation(Plan.PlanValidationType.AUTO)
+                    .build()
+            );
+        }
+    }
+
+    private void manageDocumentationPage(final Input input, final Page pageToCreate) {
+        final var pages = pageCrudService.findByApiId(input.apiId());
+        final var swaggerDocumentationPageCount = pages
+            .stream()
+            .filter(page -> page.getType() == Page.Type.SWAGGER)
+            .count();
+
+        if (swaggerDocumentationPageCount > 1) {
+            log.error("More than one Swagger documentation page already exists for this API.");
+            return;
+        }
+
+        if (swaggerDocumentationPageCount == 0) {
+            log.error("No Swagger documentation page exists for this API.");
```

Comment:
We should use a `warn` (or `debug`) because we are not failing. And we need to add the api id